tests/phpunit/CRM/Event/Import/Parser/ParticipantTest.php

```diff
@@ -88,11 +94,13 @@ protected function importCSV(string $csv, array $fieldMappings, array $submitted
     $form = $this->getFormObject('CRM_Event_Import_Form_MapField', $submittedValues);
     $form->setUserJobID($this->userJobID);
     $form->buildForm();
+    $this->assertTrue($form->validate());
```

Comment:
@demeritcowboy I managed to add these after two of the `buildForm` calls - the first one fails on the file upload rule which seems tricky

Reply:
Cool. Would need to look closer - there's that low-level php function move_uploaded_file or something that you used to have to use in the previous century to access the file for security reasons, so maybe there's an extra hurdle for those.

Reply:
yep - but running validate will def help find stuff